onchain/contracts/DAOracle.sol

```diff
@@ -9,47 +9,80 @@ import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol"
 contract DAOracle is ChainlinkClient, AccessControl {
     using Chainlink for Chainlink.Request;
 
-    mapping(uint256 => uint256) public confidence;
-    string public lightClientURL;
+    struct LightClient {
+        string url;
+        bool use;
+        bool exists;
+    }
+    struct Confidence {
+        uint256[] values;
+        uint256 max;
+        uint256 min;
+        uint256 recent;
+        bool exists;
+    }
+    mapping(uint256 => Confidence) public confidence;
+    mapping(bytes32 => LightClient) public jobs;
+    bytes32[] public jobList;
     address public oracle;
-    bytes32 public jobId;
     uint256 public fee;
     
-    event LightClientUpdated(string old_, string new_);
+    event LightClientUpdated(bytes32 jobId, string url, bool enabled);
     event OracleUpdated(address old_, address new_);
-    event JobIdUpdated(bytes32 old_, bytes32 new_);
     event FeeUpdated(uint256 old_, uint256 new_);
     event BlockConfidence(uint256 indexed block_, uint256 confidence_);
+    event BlockConfidenceRequest(uint256 indexed block_, bytes32 requestId_);
     
     constructor(address token_) {
         setChainlinkToken(token_);
-        lightClientURL = "https://polygon-da-light.matic.today/v1/confidence";
+
         oracle = 0x1cf7D49BE7e0c6AC30dEd720623490B64F572E17;
-        jobId = 'b29e1e51ae054c42849407b3cc28690d';
         fee = 10 ** 16;
+        jobs['b29e1e51ae054c42849407b3cc28690d'] = LightClient("https://polygon-da-light.matic.today/v1/confidence", true, true);
+        jobList.push('b29e1e51ae054c42849407b3cc28690d');
```

Comment:
Noob question: Why do we use a default jobId?

Reply:
It's setup step, if not done `requestConfidence` calls won't do anything useful, if `jobList` is empty

Reply:
Suggestion: jobId as param to constructor instead of fixed constant?